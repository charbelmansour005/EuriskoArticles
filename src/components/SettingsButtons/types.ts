export type FunctionProps = {
  askLogout: () => void
  navigate: (args?: any) => void
}

export type SettingsButtonsProps = FunctionProps & {
  chosenRippleColor?: string
  language?: any
}
