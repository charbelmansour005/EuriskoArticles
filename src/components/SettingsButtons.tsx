import {View, Text, StyleSheet} from 'react-native';
import {TouchableRipple} from 'react-native-paper';
import {themeColors} from '../helpers/themeColors';
import React from 'react';

type FunctionProps = {
  askLogout: () => void;
  navigate: (args?: any) => void;
};
type Props = FunctionProps & {
  chosenRippleColor?: string;
  language?: any;
};

const SettingsButtons = ({...props}: Props): JSX.Element => {
  return (
    <View style={styles.settingsBtns}>
      <TouchableRipple
        style={styles.rippleButton}
        onPress={props.askLogout}
        rippleColor={props.chosenRippleColor}>
        <Text style={styles.rippleText}>
          {props.language.english ? `Logout` : `DÃ©connecter`}
        </Text>
      </TouchableRipple>
      <TouchableRipple
        style={styles.rippleButton}
        onPress={() => props.navigate('About')}
        rippleColor={props.chosenRippleColor}>
        <Text style={styles.rippleText}>
          {props.language.english ? `About` : `Nous`}
        </Text>
      </TouchableRipple>
    </View>
  );
};

export default SettingsButtons;

const styles = StyleSheet.create({
  settingsBtns: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-around',
    alignContent: 'center',
    backgroundColor: 'white',
    margin: 10,
    padding: 10,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: 'lightgray',
    flexDirection: 'row',
  },
  rippleButton: {
    padding: 5,
    backgroundColor: themeColors.salmon,
    borderRadius: 0,
    width: '30%',
    borderWidth: 1,
    borderColor: themeColors.salmon,
  },
  rippleButtonModal: {
    padding: 4,
    borderRadius: 3,
    width: '100%',
    backgroundColor: themeColors.salmon,
  },
  rippleText: {
    textAlign: 'center',
    letterSpacing: 0,
    color: 'white',
    fontWeight: 'bold',
  },
});
