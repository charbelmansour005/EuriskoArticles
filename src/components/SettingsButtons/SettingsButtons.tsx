import {View, Text, StyleSheet} from 'react-native'
import {TouchableRipple} from 'react-native-paper'
import {themeColors} from '../../helpers/themeColors'
import React from 'react'
import {Props} from './types'

const SettingsButtons = ({...props}: Props): JSX.Element => {
  return (
    <View style={styles.settingsBtns} testID="parent">
      <TouchableRipple
        testID="logoutButton"
        style={styles.rippleButton}
        onPress={props?.askLogout}
        rippleColor={props?.chosenRippleColor}>
        <Text style={styles.rippleText}>
          {props?.language?.english ? `Logout` : `DÃ©connecter`}
        </Text>
      </TouchableRipple>
      <TouchableRipple
        testID="aboutButton"
        style={styles.rippleButton}
        onPress={() => props?.navigate('About')}
        rippleColor={props?.chosenRippleColor}>
        <Text style={styles.rippleText}>
          {props?.language?.english ? `About` : `Technologie`}
        </Text>
      </TouchableRipple>
    </View>
  )
}

export default SettingsButtons

const styles = StyleSheet.create({
  settingsBtns: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-around',
    alignContent: 'center',
    backgroundColor: 'white',
    margin: 10,
    padding: 10,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: 'lightgray',
    flexDirection: 'row',
  },
  rippleButton: {
    padding: 5,
    backgroundColor: 'seagreen',
    borderRadius: 5,
    width: '30%',
    borderWidth: 1,
    borderColor: themeColors.white,
  },
  rippleText: {
    textAlign: 'center',
    letterSpacing: 0,
    color: 'white',
    fontWeight: 'bold',
  },
})
