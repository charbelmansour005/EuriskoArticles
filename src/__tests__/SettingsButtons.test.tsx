import React from 'react'
import renderer from 'react-test-renderer'
import SettingsButtons from '../components/SettingsButtons/SettingsButtons'
// passed
test('renders correctly', async () => {
  const tree = renderer
    .create(
      <SettingsButtons
        askLogout={function (): void {
          throw new Error('Function not implemented.')
        }}
        navigate={function (args?: any): void {
          throw new Error('Function not implemented.')
        }}
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('parent view shows', async () => {
  const instance = renderer.create(
    <SettingsButtons
      askLogout={function (): void {
        throw new Error('Function not implemented.')
      }}
      navigate={function (args?: any): void {
        throw new Error('Function not implemented.')
      }}
    />,
  ).root
  expect(instance.findByProps({testID: 'parent'})).toBeTruthy()
})

test('logout button shows', async () => {
  const instance = renderer.create(
    <SettingsButtons
      askLogout={function (): void {
        throw new Error('Function not implemented.')
      }}
      navigate={function (args?: any): void {
        throw new Error('Function not implemented.')
      }}
    />,
  ).root
  expect(instance.findByProps({testID: 'logoutButton'})).toBeTruthy()
})

test('about button shows', async () => {
  const instance = renderer.create(
    <SettingsButtons
      askLogout={function (): void {
        throw new Error('Function not implemented.')
      }}
      navigate={function (args?: any): void {
        throw new Error('Function not implemented.')
      }}
    />,
  ).root
  expect(instance.findByProps({testID: 'aboutButton'})).toBeTruthy()
})
