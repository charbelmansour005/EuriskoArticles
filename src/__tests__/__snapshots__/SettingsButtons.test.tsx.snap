// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "lightgray",
      "borderRadius": 10,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "margin": 10,
      "padding": 10,
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        Object {
          "backgroundColor": "seagreen",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "padding": 5,
          "width": "30%",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "letterSpacing": 0,
          "textAlign": "center",
        }
      }
    >
      DÃ©connecter
    </Text>
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        false,
        false,
        Object {
          "backgroundColor": "seagreen",
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "padding": 5,
          "width": "30%",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
          "letterSpacing": 0,
          "textAlign": "center",
        }
      }
    >
      Technologie
    </Text>
  </View>
</View>
`;
